prometheus/vec.go

```diff
@@ -193,21 +193,40 @@ func (m *MetricVec) Delete(labels Labels) bool {
 		return false
 	}
 
-	return m.deleteByHash(h, labels)
+	return m.deleteByHashWithLabels(h, labels)
 }
 
-// deleteByHash removes the metric from the hash bucket h. If there are
-// multiple matches in the bucket, use lvs to select a metric and remove only
-// that metric.
-//
-// lvs MUST be of type Labels or []string or this method will panic.
-func (m *MetricVec) deleteByHash(h uint64, lvs interface{}) bool {
+// deleteByHashWithLabelValues removes the metric from the hash bucket h. If
+// there are multiple matches in the bucket, use lvs to select a metric and
+// remove only that metric.
+func (m *MetricVec) deleteByHashWithLabelValues(h uint64, lvs []string) bool {
 	metrics, ok := m.children[h]
 	if !ok {
 		return false
 	}
 
-	i := m.findMetric(metrics, lvs)
+	i := m.findMetricWithLabelValues(metrics, lvs)
+	if i >= len(metrics) {
+		return false
+	}
+
+	if len(metrics) > 1 {
```

Comment:
True, but I guess it would not improve readability. So I better leave it as it is.